### PR TITLE
Update cisco nxos show ip bgp neighbors

### PR DESCRIPTION
**Pull request by tom0010**
- Fix incorrect variable assignment for UPDATES_COUNT_RCVD
- Fix BGP_STATE variable order
- Update all tests as most were invalid due to UPDATES_COUNT_RCVD being incorrect
- Add support for PEER_GROUP
- Add support for LOCAL_ASN
- Add support for AS dot notation
- Add alias for REMOTE_ASN from ASN

### CODE REVIEW

**Comment by joewesch** [comment]
> Rename ASN to REMOTE_ASN to align with LOCAL_ASN

This will be a breaking change and that key name goes back 6 years.

We've done breaking changes before so it's not the worst, but maybe we should think about adding LOCAL_ASN without renaming ASN to keep it backwards compatible. Just my two cents.

**Comment by tom0010** [comment]
> > Rename ASN to REMOTE_ASN to align with LOCAL_ASN
> 
> This will be a breaking change and that key name goes back 6 years.
> 
> We've done breaking changes before so it's not the worst, but maybe we should think about adding LOCAL_ASN without renaming ASN to keep it backwards compatible. Just my two cents.

Can you let me know if that needs to be changed back or not please? It's not really a small thing to go back to `ASN` because of the order inside the test yaml files.

**Comment by joewesch** [comment]
One alternative is to record it as both ASN and REMOTE_ASN to keep backwards compatible. Turn it into a "alias" of sorts.

```
  ^BGP neighbor is ${NEIGHBOR},\s+remote AS\s+${ASN},.* -> Continue
  ^BGP neighbor is ${NEIGHBOR},\s+remote AS\s+${REMOTE_ASN},\slocal\s+AS\s+${LOCAL_ASN}.*
  ^BGP neighbor is ${NEIGHBOR},\s+remote AS\s+${REMOTE_ASN},.*
```

**Comment by tom0010** [comment]
> One alternative is to record it as both ASN and REMOTE_ASN to keep backwards compatible. Turn it into a "alias" of sorts.
> 
> ```
>   ^BGP neighbor is ${NEIGHBOR},\s+remote AS\s+${ASN},.* -> Continue
>   ^BGP neighbor is ${NEIGHBOR},\s+remote AS\s+${REMOTE_ASN},\slocal\s+AS\s+${LOCAL_ASN}.*
>   ^BGP neighbor is ${NEIGHBOR},\s+remote AS\s+${REMOTE_ASN},.*
> ```

Hey @joewesch I made these changes, please review and let me know if everything is ok.

File: ntc_templates/templates/cisco_nxos_show_ip_bgp_neighbors.textfsm [hunk 1]
```diff
@@ -84,7 +90,9 @@ Start
 AddrFamCap
   ^\s+${ADDR_FAM_ADV} -> Start
   ^BGP\s+neighbor\s+is -> Continue.Record
-  ^BGP\s+neighbor\s+is\s+${NEIGHBOR},\s+remote\s+AS\s+${ASN},.* -> Start
+  ^BGP\s+neighbor\s+is\s+${NEIGHBOR},\s+remote AS\s+${ASN},.* -> Continue
+  ^BGP\s+neighbor\s+is\s+${NEIGHBOR},\s+remote\s+AS\s+${REMOTE_ASN},\s+local\s+AS\s+${LOCAL_ASN}.* -> Start
+  ^BGP\s+neighbor\s+is\s+${NEIGHBOR},\s+remote\s+AS\s+${REMOTE_ASN},.* -> Start
```
[the remark this answers is not in the document]
Comment (tom0010) [reply ntc_templates/templates/cisco_nxos_show_ip_bgp_neighbors.textfsm]: You're right on this one, we don't need this change.

File: ntc_templates/templates/cisco_nxos_show_ip_bgp_neighbors.textfsm [hunk 2]
```diff
@@ -98,4 +106,6 @@ AddrFamState
   ^\s+Foreign host:\s+${REMOTE_IP}, Foreign port:\s+${REMOTE_PORT}
   ^\s+No\s+established\s+BGP\s+session\s+with\s+peer
   ^BGP\s+neighbor\s+is -> Continue.Record
-  ^BGP\s+neighbor\s+is\s+${NEIGHBOR},\s+remote\s+AS\s+${ASN},.* -> Start
+  ^BGP\s+neighbor\s+is\s+${NEIGHBOR},\s+remote AS\s+${ASN},.* -> Continue
+  ^BGP\s+neighbor\s+is\s+${NEIGHBOR},\s+remote\s+AS\s+${REMOTE_ASN},\s+local\s+AS\s+${LOCAL_ASN}.* -> Start
+  ^BGP\s+neighbor\s+is\s+${NEIGHBOR},\s+remote\s+AS\s+${REMOTE_ASN},.* -> Start
```
[the remark this answers is not in the document]
Comment (tom0010) [reply ntc_templates/templates/cisco_nxos_show_ip_bgp_neighbors.textfsm:109]: I can't get these tests to pass without adding these new lines.

[the remark this answers is not in the document]
Comment (joewesch) [reply ntc_templates/templates/cisco_nxos_show_ip_bgp_neighbors.textfsm:109]: Ah, I was incorrect. The state transition [does process the line like normal and then changes states afterwards](https://github.com/google/textfsm/wiki/TextFSM#new-state-transition).

Let's at least change all spaces to `\s+` consistency, please:
```
  ^BGP\s+neighbor\s+is\s+${NEIGHBOR},\s+remote\s+AS\s+${ASN},.* -> Continue
```

[the remark this answers is not in the document]
Comment (tom0010) [reply ntc_templates/templates/cisco_nxos_show_ip_bgp_neighbors.textfsm]: Fixed

**Review by jvanderaa — APPROVED** [review]

**Pull request merged**